kernel/aster-nix/src/vm/vmar/vm_mapping.rs

```diff
@@ -468,11 +468,13 @@ impl VmMappingInner {
         let vmo_map_range = (range.start - map_to_addr + self.vmo_offset)
             ..(range.end - map_to_addr + self.vmo_offset);
         let page_idx_range = get_page_idx_range(&vmo_map_range);
-        for page_idx in page_idx_range {
-            self.unmap_one_page(vm_space, page_idx)?;
+        let original_mapped_pages = self.mapped_pages.clone();
+        let mapped_pages_in_range = original_mapped_pages.range(page_idx_range);
```

Comment:
The `clone` can be avoided here, right? The `BtreeSet::range` method accepts `&self` as parameter.

Reply:
This is because `unmap_one_page` need a mutable reference to self and it will change the `mapped_pages` field. So I think it needs to record the original `mapped_pages`.

Reply:
Got it! It seems that we need a more efficient API for unmapping range instead of unmapping one page at a time. But this may be beyond the scope of this PR.